proposals/src/traits.rs

```diff
@@ -0,0 +1,14 @@
+use crate::Proposal;
+
+/// Trait to be used for handling signed proposals
+pub trait OnSignedProposal<E> {
+    /// On a signed proposal, this method is called.
+    /// It returns a result `()` and otherwise an error of type `E`.
+    fn on_signed_proposal(proposal: Proposal) -> Result<(), E>;
+}
```

Comment:
Is there any reason why we need this trait here? in this crate. I want to keep this crate lightweight as possible and `no_std` compatible (for being used in other things like wasm based contracts ...etc)

Reply:
Yes, this trait is used now by both `protocol-substrate` and `dkg-substrate`. It also is super lightweight and has no issues with `no_std` compatibility. We must always ensure everything is no_std ofc.